Documentation/Functions/Data.rst

```diff
@@ -797,6 +797,45 @@ Get content from the :ref:`register <using-setting-register>`:
 
     lib.foo.data = register : color
 
+
+..  _data-type-gettext-request:
+
+request
+-------
+
+..  versionadded:: 12.4
+
+..  confval:: request
+    :name: data-request
+    :Syntax: request : [attribute] | [property]
+
+    Retrieve the property of a
+    :ref:`PSR-7 request attribute <t3coreapi:request-attributes>`.
+
+    Note that only scalar properties can be retrieved: int, float, string or
+    bool.
+
+..  _data-type-gettext-request-example-page-type:
+
+Example: Get the page type
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  code-block:: typoscript
+    :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+
+    lib.foo.data = request : routing | pageType
```

Comment:
I haven't much used this, so please bear with me, regard me as the target audience. ;)

- The syntax looks a bit to me as if "|" might be an "or" syntax. Maybe we should note that it's not to be read like "use an attribute OR a property" but "first retrieve the attribute, then a soecific property of that attribut"

- The current docs in "request attributes" don't really seem to list available properties for each attribute, did I read that right? Like for the nonce, how do I know that "value" is what I need, if it wasn't this precise example? Same for "pageType" of "routing" - https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/Routing.html#routing does not have an entry "pageType". But it does have a getter, which not everyone might directly understand. This docs here could state that "access every property that has a getter listed in the documentation; for getPageType you can use "pageType" as property, and NOT "getPageType". Or on the other docs we could offer a simple list of "property" values, so one doesn't have to scroll through a large list of methods...?

Reply:
1) The "|" is common in TS, see also here: https://docs.typo3.org/m/typo3/reference-typoscript/12.4/en-us/Functions/Data.html#session. So, when explaining, that should be done for all key/value pairs which is out of scope of this PR.

2) The code looks for getters or properties. If they are available and return a scalar, they are returned. If they are an array or object, an empty string is returned (yes, have to look into the code for that). But I agree, it can be explained better. I don't like the idea of a property list in the attributes chapter, as these properties are mostly private/protected and so of little use for the average user of request attributes.

I also played around a bit with those TS configuration. IMHO the only usable attributes are currently "routing" with pageType and pageId and "nonce". One can also use "site" or "language" but we have dedicated data functions for that, which are easier to use. Same for TSFE.

The request data function has been introduced because of the deprecation of TSFE->type in v12.4 and is therefore not available anymore in v13.0 (as "TSFE : type").